### PR TITLE
Honor radio's option to use Fahrenheit.

### LINKED ISSUE #599: Feature Request: Display Fahrenheit on Android
The Android app only displays temperature in celsius.  Request a feature to be added to allow users to select between fahrenheit and celsius.

### PR DESCRIPTION
**Pull request by jp-bennett**
Closes #599

### CODE REVIEW

File: app/src/main/java/com/geeksville/mesh/NodeInfo.kt [hunk 1]
```diff
@@ -174,13 +174,27 @@ data class NodeInfo(
...
+                envFormat("%.1f", "mA", environmentMetrics?.current) }
+        else {
+            return envFormat("%.1f", "°C ", environmentMetrics?.temperature) +
                 envFormat("%.0f", "%% ", environmentMetrics?.relativeHumidity) +
                 envFormat("%.1f", "hPa ", environmentMetrics?.barometricPressure) +
                 envFormat("%.0f", "mΩ ", environmentMetrics?.gasResistance) +
                 envFormat("%.2f", "V ", environmentMetrics?.voltage) +
                 envFormat("%.1f", "mA", environmentMetrics?.current)
+        }
+    }
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/NodeInfo.kt]:
```suggestion
    fun envMetricStr(isFahrenheit: Boolean = false): String =
        if (!isFahrenheit) envFormat("%.1f", "°C ", environmentMetrics?.temperature)
        else envFormat("%.1f", "°F ", environmentMetrics?.temperature?.times(1.8f)?.plus(32)) +
                envFormat("%.0f", "%% ", environmentMetrics?.relativeHumidity) +
                envFormat("%.1f", "hPa ", environmentMetrics?.barometricPressure) +
                envFormat("%.0f", "mΩ ", environmentMetrics?.gasResistance) +
                envFormat("%.2f", "V ", environmentMetrics?.voltage) +
                envFormat("%.1f", "mA", environmentMetrics?.current)
```

Comment (andrekir) [reply app/src/main/java/com/geeksville/mesh/NodeInfo.kt]: - referencing Fahrenheit in the parameter name to improve readability
- you can avoid repetition using only the temperature string inside the if/else block:

Celsius: `envFormat("%.1f", "°C ", environmentMetrics?.temperature)`
Fahrenheit: `envFormat("%.1f", "°F ", environmentMetrics?.temperature?.times(1.8f)?.plus(32))`

File: app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt [hunk 2]
```diff
@@ -205,8 +205,8 @@ class UsersFragment : ScreenFragment("Users"), Logging {
 
             holder.lastTime.text = formatAgo(n.lastHeard)
 
-            if (n.envMetricStr.isNotEmpty()) {
-                holder.envMetrics.text = n.envMetricStr
+            if (n.envMetricStr(model.module.telemetry.environmentDisplayFahrenheit).isNotEmpty()) {
+                holder.envMetrics.text = n.envMetricStr(model.module.telemetry.environmentDisplayFahrenheit)
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt]: make sure to check your added code follows [coding conventions](https://kotlinlang.org/docs/coding-conventions.html). to reformat the selected text In Android Studio there is the Ctrl + Alt + L hotkey.

you could also store `n.envMetricStr(model.module.telemetry.environmentDisplayFahrenheit` in a variable before the if/else block so it is only called once.

**Review by andrekir — CHANGES_REQUESTED** [review]

**Comment by jp-bennett** [comment]
@andrekir I knew there would be some Kotlin magic I missed out on. Thanks for the suggestions, I'll test and amend the PR in a bit.

**Comment by jp-bennett** [comment]
@andrekir Thanks again. I tried the .times()?.plus() construction during development, and was getting weird results. But apparently that was spurious and unrelated, as this morning it's working just fine. I think this is ready to go.

**Comment by andrekir** [comment]
looks good. thanks!

**Pull request merged**